### DISCUSSION TITLE
Keys

### DISCUSSION THREAD

**Discussion by ppochez** [Artificial-Pancreas/iAPS#532] [Q&A]
Hi, I'm following the steps to build iAPS and am stuck at making a key! 
When I register a new key, it doesn't look like how it does in the loop video tutorial. Instead, I'm driven to this page. (option to enable Apple Push Notifications service (APNs), DeviceCheck, Account & Organizational Data Sharing, ClassKit Catalog, WeatherKit. 
Not sure how to move forward. Any help appreciated! 
<img width="1186" alt="Screen Shot 2024-02-14 at 12 18 25" src="https://github.com/Artificial-Pancreas/iAPS/assets/156239901/a127fac2-1c0f-4196-a1c0-f58a205bff56">

**Comment by straykatz** [comment]
Hi, I have not done this myself, but here is what I found about APN keys: 

https://developer.apple.com/documentation/usernotifications/registering_your_app_with_apns

At the link, there is a description how to enable an app to create a APN key in xcode. 

I hope this helps...  assume this requires using Xcode for building your App. 
Cheers. K

![image](https://github.com/Artificial-Pancreas/iAPS/assets/45725845/070c5801-79ed-43e2-af61-eaea2c3b8562)
